### ISSUE TITLE
Sync is still broken on Kubuntu 22.04.2 LTS 64-bit (Brave v151.114)

### ISSUE THREAD

**Issue by NeilPandya** [brave/brave-browser#30249]
## Description 

Still receiving the following error:

```
Sync service could not be started, Brave Sync is not running
```
![Screenshot_20230510_085804](https://github.com/brave/brave-browser/assets/21351020/39f828a1-072e-4e5f-804e-7bbcc09b8bd8)

![Screenshot_20230510_085837](https://github.com/brave/brave-browser/assets/21351020/cbe7cafc-4250-49fc-8abe-39c491964b56)

Cross-referencing this issue with #30147, which I assumed to be solved with PR[#18404](https://github.com/brave/brave-core/pull/18404), but still not able to sync.

Rolled back to v150.125 via:
```bash
$ sudo apt install brave-browser=150.125
```
Sync still works on this version (150.125).

## Steps to Reproduce

   1. Install Brave
```bash
$ sudo apt install curl

$ sudo curl -fsSLo /usr/share/keyrings/brave-browser-archive-keyring.gpg https://brave-browser-apt-release.s3.brave.com/brave-browser-archive-keyring.gpg

$ echo "deb [signed-by=/usr/share/keyrings/brave-browser-archive-keyring.gpg] https://brave-browser-apt-release.s3.brave.com/ stable main"|sudo tee /etc/apt/sources.list.d/brave-browser-release.list

$ sudo apt update

$ sudo apt install brave-browser
```
   2. Open Brave
```bash
$ brave-browser
```
   3. Try to enter sync-phrase.

## Actual result:

```
Sync service could not be started, Brave Sync is not running
```

## Expected result:
Synced bookmarks, passwords, extensions, etc.

## Reproduces how often: 

Easily reproduced

## Brave version (brave://version info)

[Version 1.51.114 Chromium: 113.0.5672.92 (Official Build) (64-bit)](https://brave.com/latest/)

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes.
- Can you reproduce this issue with the beta channel? No.
- Can you reproduce this issue with the nightly channel? No.

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No
- Does the issue resolve itself when disabling Brave Rewards? No
- Is the issue reproducible on the latest version of Chrome? N/A. As far as I'm aware, Chrome doesn't use the same syncing system as Brave.

## Miscellaneous Information:
 
Related Posts, Issues, and Pull Requests:
[Community](https://community.brave.com/t/sync-not-working-brave-was-unable-to-access-your-systems-secure-storage/485452)
#30147
#30167
[PR#18404](https://github.com/brave/brave-core/pull/18404)

Tried to resolve with @AlexeyBarabash's and @Rad0van's suggestions detailed [here](https://github.com/brave/brave-browser/issues/30147#issuecomment-1540807353), but it didn't work.

My Specs:
```bash
           `.:/ossyyyysso/:.               neil@FractalNorth 
        .:oyyyyyyyyyyyyyyyyyyo:`           ----------------- 
      -oyyyyyyyodMMyyyyyyyysyyyyo-         OS: Kubuntu 22.04.2 LTS x86_64 
    -syyyyyyyyyydMMyoyyyydmMMyyyyys-       Kernel: 5.19.0-41-generic 
   oyyysdMysyyyydMMMMMMMMMMMMMyyyyyyyo     Uptime: 1 hour, 9 mins 
 `oyyyydMMMMysyysoooooodMMMMyyyyyyyyyo`    Packages: 2501 (dpkg), 18 (flatpak), 7 (snap) 
 oyyyyyydMMMMyyyyyyyyyyyysdMMysssssyyyo    Shell: bash 5.1.16 
-yyyyyyyydMysyyyyyyyyyyyyyysdMMMMMysyyy-   Resolution: 1920x1080 
oyyyysoodMyyyyyyyyyyyyyyyyyyydMMMMysyyyo   DE: Plasma 5.24.7 
yyysdMMMMMyyyyyyyyyyyyyyyyyyysosyyyyyyyy   WM: KWin 
yyysdMMMMMyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy   WM Theme: Sweet-Dark-transparent 
oyyyyysosdyyyyyyyyyyyyyyyyyyydMMMMysyyyo   Theme: [Plasma], Breeze [GTK2/3] 
-yyyyyyyydMysyyyyyyyyyyyyyysdMMMMMysyyy-   Icons: candy-icons [Plasma], candy-icons [GTK2/3] 
 oyyyyyydMMMysyyyyyyyyyyysdMMyoyyyoyyyo    Terminal: konsole 
 `oyyyydMMMysyyyoooooodMMMMyoyyyyyyyyo     CPU: AMD Ryzen 9 5950X (32) @ 3.400GHz 
   oyyysyyoyyyysdMMMMMMMMMMMyyyyyyyyo      GPU: NVIDIA GeForce RTX 3090 
    -syyyyyyyyydMMMysyyydMMMysyyyys-       Memory: 2757MiB / 32008MiB 
      -oyyyyyyydMMyyyyyyysosyyyyo-
        ./oyyyyyyyyyyyyyyyyyyo/.                                   
           `.:/oosyyyysso/:.`                                      
```

**Comment by Rad0van** [comment]
I wasn't using Sync at all actually. Don't know whether it's just coincidence that after upgrade of Brave I lost access to the stored passwords (but that's part of symptoms for others as well)

**Comment by NeilPandya** [comment]
> I wasn't using Sync at all actually. Don't know whether it's just coincidence that after upgrade of Brave I lost access to the stored passwords (but that's part of symptoms for others as well)

I have a feeling issues with decrypting the KWallet key(s) are at the heart of the problem.

**Comment by AlexeyBarabash** [comment]
@NeilPandya 
The fix is available on Nightly since `1.53.27`

For Stable channel it will be available with `1.51.x` maintenance release that's scheduled for May 17, 2023.

**Comment by NeilPandya** [comment]
> @NeilPandya 
> The fix is available on Nightly since `1.53.27`
> 
> For Stable channel it will be available with `1.51.x` maintenance release that's scheduled for May 17, 2023.

Thanks, I will close the issue, as I should've just read the release notes.
